src/composable/ConditionalOrder.ts

```diff
@@ -189,23 +213,125 @@ export abstract class ConditionalOrder<Data, Params> {
    */
-  protected encodeStaticInputHelper(orderDataTypes: string[], staticInput: Data): string {
+  protected encodeStaticInputHelper(orderDataTypes: string[], staticInput: S): string {
     return utils.defaultAbiCoder.encode(orderDataTypes, [staticInput])
   }
 
   /**
-   * Apply any transformations to the parameters that are passed in to the constructor.
+   * Poll a conditional order to see if it is tradeable.
+   *
+   * @param owner The owner of the conditional order.
+   * @param p The proof and parameters.
+   * @param chain Which chain to use for the ComposableCoW contract.
+   * @param provider An RPC provider for the chain.
+   * @param offChainInputFn A function, if provided, that will return the off-chain input for the conditional order.
+   * @throws If the conditional order is not tradeable.
+   * @returns The tradeable `GPv2Order.Data` struct and the `signature` for the conditional order.
+   */
+  async poll(owner: string, chain: SupportedChainId, provider: providers.Provider): Promise<PollResult> {
+    const composableCow = getComposableCow(chain, provider)
+
+    try {
+      const isValid = this.isValid()
+      // Do a validation first
+      if (!isValid.isValid) {
+        return {
+          result: PollResultCode.DONT_TRY_AGAIN,
+          reason: `InvalidConditionalOrder. Reason: ${isValid.reason}`,
+        }
+      }
+
+      // Let the concrete Conditional Order decide about the poll result
+      const pollResult = await this.pollValidate(owner, chain, provider)
+      if (pollResult) {
+        return pollResult
+      }
+
+      // Check if the owner authorised the order
+      const isAuthorized = await this.isAuthorized(owner, chain, provider)
+      if (!isAuthorized) {
+        return {
+          result: PollResultCode.DONT_TRY_AGAIN,
+          reason: `NotAuthorised: Order ${this.id} is not authorised for ${owner} on chain ${chain}`,
+        }
+      }
+
+      // Lastly, try to get the tradeable order and signature
+      const [order, signature] = await composableCow.getTradeableOrderWithSignature(
+        owner,
+        this.leaf,
+        this.offChainInput,
+        []
+      )
+
+      return {
+        result: PollResultCode.SUCCESS,
+        order,
+        signature,
+      }
+    } catch (error) {
+      return {
+        result: PollResultCode.UNEXPECTED_ERROR,
+        error: error,
+      }
+    }
+  }
+
+  /**
+   * Checks if the owner authorized the conditional order.
+   *
+   * @param owner The owner of the conditional order.
+   * @param chain Which chain to use for the ComposableCoW contract.
+   * @param provider An RPC provider for the chain.
+   * @returns true if the owner authorized the order, false otherwise.
+   */
+  public isAuthorized(owner: string, chain: SupportedChainId, provider: providers.Provider): Promise<boolean> {
+    const composableCow = getComposableCow(chain, provider)
+    return composableCow.callStatic.singleOrders(owner, this.id)
```

Comment:
To verify the proof, one would calculate the root from the proof given, then verify this against the `root` stored in `composableCow` for that owner.